packages/e2e-tests/src/Tar.ts

```diff
@@ -0,0 +1,133 @@
+import { absPath, command, cwd, log, logSpawnSyncError } from './utilities';
+import child_process from 'node:child_process';
+import fs from 'node:fs';
+
+class Tar {
+	private readonly group: string = 'tar';
+
+	private isInstalled(): boolean {
+		return command('tar', this.group);
+	}
+
+	/**
+	 * @param files Files to be tarballed
+	 * @param tarball Path to the (newly) created tarball
+	 * @returns Absolute path to tarball file or `false` on failure
+	 */
+	public create(files: string | string[], tarball?: string): string | false {
+		if (!this.isInstalled()) {
+			return false;
+		}
+
+		const inputs = typeof files === 'string' ? [files] : files;
+
+		for (const i of inputs) {
+			if (!fs.existsSync(i)) {
+				log(`Given tar input '${i}' does not exist!`, { group: this.group });
+				return false;
+			}
+		}
+
+		const output = this.getTarballPath(files, tarball);
+
+		if (!output) {
+			return false;
+		}
+
+		if (fs.existsSync(output)) {
+			log(`Output path for tarball already exists: '${output}' !`, { group: this.group });
+			return false;
+		}
+
+		const command = child_process.spawnSync('tar', ['--create', '--verbose', '--file', output, ...inputs], {
+			stdio: 'pipe',
+			encoding: 'utf-8',
+		});
+
+		console.log(command.stdout);
+
+		if (command.status !== 0) {
+			logSpawnSyncError({ command, group: this.group, message: 'Could not create tarball!' });
+			return false;
+		}
+
+		return output;
+	}
+
+	private getTarballPath(files: string | string[], archive?: string): string | false {
+		if (archive && archive.length === 0) {
+			log('Given empty string to tar output path!', { group: this.group });
+			return false;
+		}
+
+		if (typeof files === 'string') {
+			if (archive) {
+				return absPath(archive);
+			}
+			if (!archive) {
+				return files.replace(/\/$/, '') + '.tar';
+			}
+		}
+
+		if (Array.isArray(files)) {
+			if (archive) {
+				return absPath(archive);
+			}
+			if (!archive) {
```

Comment:
is the second conditional `if (!archive) {` necessary since it follows `if (archive) {`?

Reply:
True, I'll update the code.